pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -371,18 +373,24 @@ def on_validation_end(self, trainer, pl_module):
             elif self.check_monitor_top_k(current):
                 self._do_check_save(filepath, current, epoch, trainer, pl_module)
             elif self.verbose > 0:
-                log.info(f'\nEpoch {epoch:05d}: {self.monitor}  was not in top {self.save_top_k}')
+                log.info(f'Epoch {epoch:d}: {self.monitor} was not in top {self.save_top_k}')
 
         else:
             if self.verbose > 0:
-                log.info(f'\nEpoch {epoch:05d}: saving model to {filepath}')
+                log.info(f'Epoch {epoch:d}: saving model to {filepath}')
 
             assert trainer.global_rank == 0, 'tried to make a checkpoint from non global_rank=0'
             self._save_model(filepath, trainer, pl_module)
 
         if self.save_last:
-            filepath = os.path.join(self.dirpath, self.prefix + ModelCheckpoint.CHECKPOINT_NAME_LAST)
+            filename = self._format_checkpoint_name(
+                self.CHECKPOINT_NAME_LAST, epoch, ckpt_name_metrics, prefix=self.prefix
+            )
+            filepath = os.path.join(self.dirpath, f'{filename}.ckpt')
+            if self.last_model_path:
+                self._del_model(self.last_model_path)
             self._save_model(filepath, trainer, pl_module)
```

Comment:
```suggestion
            self._save_model(filepath, trainer, pl_module)
            if self.last_model_path and self.last_model_path != filepath:
                self._del_model(self.last_model_path)
```
I think it should save first and then delete the old one.

Reply:
That does not work when you have a checkpoint from a previous run with the same name.

Say you already have a last checkpoint from a previous run named "last.ckpt" (default name). When you run your experiment again, your change would overwrite the previous "last.ckpt" and then delete it, losing both checkpoints in the process.

Reply:
You are correct. Changed!